grype/presenter/config.go

```diff
@@ -0,0 +1,44 @@
+package presenter
+
+import "fmt"
+
+// Config is the presenter domain's configuration data structure.
+type Config struct {
+	format           format
+	templateFilePath string
+}
+
+// ValidatedConfig returns a new, validated presenter.Config. If a valid Config cannot be created using the given input,
+// an error is returned.
+func ValidatedConfig(output, outputTemplateFile string) (Config, error) {
```

Comment:
nit/question: why not name this `NewConfig` instead?

Reply:
I wanted to signal that config validation was occurring here, as opposed to a more transparent constructor whose job is "create an object and use these values". I started with `NewConfig` but wanted to express to the consumer that this function tackles a noteworthy step in the flow of config into the domain. That being said, I'm not married to it. Are there reasons for `NewConfig` we should consider here?

Reply:
I would think that a constructor can do validation, and not using the "NewXXX" convention would make it harder (although not horribly so) to discover. No strong opinions here, but is seems that @luhring needs some reassurance. I'll defer to @wagoodman